lib/puppet_litmus/serverspec.rb

```diff
@@ -229,10 +254,11 @@ def targeting_localhost?
   #
   # @param command [String] The puppet command causing the error.
   # @param result  [Array] The result struct containing the result
-  def report_puppet_apply_error(command, result)
+  def report_puppet_apply_error(command, result, acceptable_exit_codes)
     puppet_apply_error = <<-ERROR
 apply manifest failed
 `#{command}`
+with exit code #{result.first['result']['exit_code']} (expected: #{acceptable_exit_codes})
 ======Start output of failed Puppet run======
```

Comment:
I think the verbage of the header /footer should be fixed. It is ambiguous, in that it indicated failure ( it may have been a change).
 It is not a puppet run, it is a puppet apply